mlflow/data/dataset.py

```diff
@@ -0,0 +1,124 @@
+import json
+from abc import abstractmethod
+from typing import Optional, Any, Dict
+
+from mlflow.data.dataset_source import DatasetSource
+from mlflow.entities import Dataset as DatasetEntity
+from mlflow.utils.annotations import experimental
+
+
+@experimental
+class Dataset:
```

Comment:
```suggestion
class Dataset(ABC):
```

I think `Dataset` should inherit `ABC` because it's an abstract class.